include/swift/SILOptimizer/Utils/InstOptUtils.h

```diff
@@ -587,6 +587,14 @@ FullApplySite cloneFullApplySiteReplacingCallee(FullApplySite applySite,
                                                 SILValue newCallee,
                                                 SILBuilderContext &builderCtx);
 
+/// Analyze the destructor for the class of \p value to see if any instructions
+/// in it could have side effects on the program outside the destructor.
```

Comment:
Can you add something like "Conservatively returns true if the destructor cannot be analyzed or if value is not a class"